app/javascript/controllers/transcript_controller.js

```diff
@@ -15,25 +15,77 @@ export default class extends Controller {
   // We can't load right away, because the VTT tracks may not have been parsed yet. 
   // This function is triggered by the 'media-data-loaded' event which is triggered
   // by the 'loadeddata' event on the first track.  
-  load() {
+  async load() {
     // Return if this method has already been called, there are no caption tracks
-    // or no cues for the tracks
-    if (this.loaded || !this.currentCues()) 
+    // or no cues for the tracks.  In the case of Safari, we need to wait to check,
+    // hence the async/await combination for checkCues.
+    if (this.loaded || !(await this.checkCues()))  
       return
-
+    
     this.revealButton()
     this.setupTranscriptLanguageSwitching()
     this.renderCues()
     this.loaded = true
   }
 
+  // Safari cues require special handling.
+  // We want the track cues to be available so we can properly generate the transcript sidebar language dropdown
+  // if there is more than one language track.
+  convertDisabledTracks() {
+    const captions = this.player.remoteTextTracks()?.tracks_.filter(track => track.kind === 'captions')
+    let hiddenTracks = false
```

Comment:
I don't see where this variable is used. Maybe cruft?

Reply:
Agreed. Removed.